src/generic/MPoly.jl

```diff
@@ -79,7 +79,16 @@ Return the $i$-th generator (variable) of the given polynomial
 ring.
 """
 function gen(a::MPolyRing{T}, i::Int) where {T <: RingElement}
-   return gen(a, i, Val(a.ord))
+   ord = internal_ordering(a)
+   if ord == :lex
+      return gen(a, i, Val(:lex))
+   elseif ord == :deglex
+      return gen(a, i, Val(:deglex))
+   elseif ord == :degrevlex
+      return gen(a, i, Val(:degrevlex))
+   else
+      error("invalid ordering")
+   end
```

Comment:
How about we do
```suggestion
   return gen(a, i, internal_ordering_as_val(a))
```
and then
```julia
function internal_ordering_as_val(a:: MPolyRing)
   ord = internal_ordering(a)
   if ord == :lex
      return Val(:lex)
   elseif ord == :deglex
      return Val(:deglex)
   elseif ord == :degrevlex
      return Val(:degrevlex)
   else
      error("invalid ordering")
   end
end
```
and then use that in the other functions, too?

Reply:
Doesn't this reintroduce runtime dispatch? I might be missing something.

Reply:
I think so. But I will try to circumvent that with forced inlining or a macro

Reply:
If it has to be fancy, I vote for a macro.

Reply:
Before we go down that route, I think it'd be better to consider just re-integrating the various functions and avoid the use of `Val` entirely.

I'll push a proposal

Reply:
Fine for me as well (if someone reproduces the benchmark from https://github.com/Nemocas/AbstractAlgebra.jl/issues/2021#issuecomment-2702097207 with the new changes here).

And couly you please also removed everything from `deprecations.jl` that references a now removed function?